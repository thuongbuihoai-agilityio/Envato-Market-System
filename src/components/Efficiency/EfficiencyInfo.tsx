import { memo } from 'react';
import { Box, Flex, Text, useColorMode } from '@chakra-ui/react';
import Chart from 'react-apexcharts';

// Components
import Statistical from './Statistical';

// Constants
import { STROKE_COLORS, THEMES } from '@app/constants';

// Icons
import { Sort } from '@app/components/Icons';

// Types
import { IEfficiency } from '@app/interfaces';

// Colors
import { colors } from '@app/themes/bases/colors';

// Utils
import { formatDecimalNumber } from '@app/utils';

const EfficiencyComponent = ({
  statistical,
  arrival,
  spending,
}: IEfficiency): JSX.Element => {
  const { colorMode } = useColorMode();

  const sortIconColor =
    colorMode === THEMES.LIGHT
      ? colors.text.primary.default
      : colors.common.white;

  return (
    <Box py={4} px={5}>
      <Flex justifyContent="space-between" mb={4}>
        <Chart
          options={{
            plotOptions: {
              pie: {
                donut: {
                  size: '45%',
                },
              },
            },
            legend: {
              show: false,
            },

            annotations: {},
            colors: STROKE_COLORS,
            dataLabels: {
              enabled: true,
              formatter: (val) => val + '%',
            },
            tooltip: {
              custom: function ({ series, seriesIndex }) {
                return `<div style="padding: 10px; background-color: #000" >
              <span>
              ${statistical[seriesIndex].title}: ${series[seriesIndex]}%
              </span>
              </div>`;
              },
            },
          }}
          series={statistical.map((item) => item.value)}
          type="donut"
          width="200"
        />
        <Box>
          <Box mb={6}>
            <Flex alignItems="center" gap={1}>
              <Text variant="textLg" color="primary.500">
                ${formatDecimalNumber(arrival, true)}
              </Text>
              <Sort />
            </Flex>
            <Text variant="textMd" color="text.secondary">
              Arrival
            </Text>
          </Box>
          <Box>
            <Flex alignItems="center" gap={1}>
              <Text variant="textLg">
                ${formatDecimalNumber(spending, true)}
              </Text>
              <Sort color={sortIconColor} />
            </Flex>

            <Text variant="textMd" color="text.secondary">
              Spending
            </Text>
          </Box>
        </Box>
      </Flex>
      <Flex flexDirection="column" gap={1.5}>
        {statistical.map((item, index) => (
          <Statistical
            key={item.title}
            {...item}
            color={STROKE_COLORS[index]}
          />
        ))}
      </Flex>
    </Box>
  );
};

const EfficiencyInfo = memo(EfficiencyComponent);

export default EfficiencyInfo;
