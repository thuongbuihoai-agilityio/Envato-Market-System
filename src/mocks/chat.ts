// Constants
import { AVATAR_POSITION, IMAGES } from '@app/constants';
import { MESSAGE_TIME } from '.';

// TODO: Update later
export const USER_CHATS = [
  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hi, I need more information',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 1000,
  },

  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hello, I want to know more about your product',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 3000,
  },

  {
    uid: 'user',
    isSend: false,
    isAudio: false,
    content: 'Hello, I want to know more about your product',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 6000,
  },

  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Hi, i am here for help',
    avatarPosition: AVATAR_POSITION.AFTER,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 2000,
  },

  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Sure, I can help you with that',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 4000,
  },
  {
    uid: 'admin',
    isSend: true,
    isAudio: false,
    content: 'Sure, I can help you with that',
    avatarPosition: AVATAR_POSITION.BEFORE,
    avatar: IMAGES.CHAT_USER_AVATAR.url,
    localeTime: MESSAGE_TIME + 7000,
  },
];

export const MEMBER_CHATS = [
  { uid: 'admin', name: 'Admin' },
  { uid: 'user1', name: 'User 1' },
];
