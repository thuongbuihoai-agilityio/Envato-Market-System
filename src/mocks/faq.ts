export const FAQ_DATA = [
  {
    id: 1,
    question: 'What is the most important part of a dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 2,
    question: 'What are the three types of dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 3,
    question: 'What are examples of admin?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 4,
    question: 'What are 5 benefits of dashboards?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 5,
    question: 'What makes a good dashboard?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
  {
    id: 6,
    question: 'What are the 3 basic skills of administrator?',
    answer:
      'In reality, the most important aspect of a great dashboard is the part that gets the least amount of attention: The underlying data. More than any other aspect, the data will make or break a dashboard.Within this definition, successful administration appears to rest on three basic skills, which we will call technical, human, and conceptual.Dashboards are business intelligence.',
  },
];
