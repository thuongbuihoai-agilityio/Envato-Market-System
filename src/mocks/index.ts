export * from './spending';
export * from './select';
export * from './table';
export * from './sidebar';
