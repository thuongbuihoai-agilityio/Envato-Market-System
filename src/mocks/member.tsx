import { Image } from '@chakra-ui/react';

// Constants
import { IMAGES } from '@app/constants';

// Utils
import { MESSAGE_TIME_FORMAT } from '.';

export const MEMBER_CHAT = [
  {
    id: 'user1',
    avatar: 'https://example.com/avatar1.jpg',
    name: 'Kien Nguyen',
    status: 'Online',
    localeTime: MESSAGE_TIME_FORMAT,
    icon: (
      <Image
        w={5}
        h={5}
        src="icons/file.svg"
        alt={IMAGES.ATTACH.alt}
        fallbackSrc={IMAGES.FALLBACK.url}
      />
    ),
    statusColor: 'online',
  },
  {
    id: 'user2',
    avatar: 'https://example.com/avatar2.jpg',
    name: 'Huy Pham',
    status: 'Away',
    localeTime: MESSAGE_TIME_FORMAT,

    icon: (
      <Image
        w={5}
        h={5}
        src="icons/file.svg"
        alt={IMAGES.ATTACH.alt}
        fallbackSrc={IMAGES.FALLBACK.url}
      />
    ),
    statusColor: 'offline',
  },
];
