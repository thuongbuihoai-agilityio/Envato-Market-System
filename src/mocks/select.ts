export const SOCIAL_PLATFORM_OPTIONS = [
  {
    value: 'github',
    label: 'github',
  },
  {
    value: 'facebook',
    label: 'facebook',
  },
  {
    value: 'instagram',
    label: 'instagram',
  },
];
