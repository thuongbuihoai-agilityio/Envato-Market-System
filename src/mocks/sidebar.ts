// constants
import {
  HELP_ITEM_LIST,
  MENU_ITEM_LIST,
  OTHER_ITEM_LIST,
} from '@app/constants/sidebar';

export const EXPAND_SIDEBAR_MENU_LIST = [
  {
    id: 1,
    title: 'Menu',
    listItem: MENU_ITEM_LIST,
  },
  {
    id: 2,
    title: 'Help',
    listItem: HELP_ITEM_LIST,
  },
  {
    id: 3,
    title: 'Others',
    listItem: OTHER_ITEM_LIST,
  },
];
