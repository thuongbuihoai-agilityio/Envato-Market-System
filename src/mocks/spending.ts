export const TOTAL_EARNINGS_MOCK = {
  title: 'Total earnings',
  total: 7.245,
  growth: 3.5,
  weeklyIncome: [30, 40, 45, 50, 49, 60, 91],
};
