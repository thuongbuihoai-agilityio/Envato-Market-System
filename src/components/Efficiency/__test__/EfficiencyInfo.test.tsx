import { render } from '@testing-library/react';
import * as chakra from '@chakra-ui/react';

// Components
import EfficiencyInfo from '../EfficiencyInfo';

// Mocks
import { EFFICIENCY_MOCK } from '@app/mocks';

jest.mock('react-apexcharts', () => ({
  __esModule: true,
  default: () => <div>EfficiencyInfo component</div>,
}));

jest.mock('@chakra-ui/react');

describe('EfficiencyInfo component', () => {
  it('should render correctly', () => {
    jest.spyOn(chakra, 'useColorMode').mockReturnValue({
      colorMode: 'light',
      setColorMode: jest.fn(),
      toggleColorMode: jest.fn(),
    });
    const { container } = render(<EfficiencyInfo {...EFFICIENCY_MOCK} />);

    expect(container).toMatchSnapshot();
  });

  it('should render with dark mode ', () => {
    jest.spyOn(chakra, 'useColorMode').mockReturnValue({
      colorMode: 'dark',
      setColorMode: jest.fn(),
      toggleColorMode: jest.fn(),
    });
    const { container } = render(<EfficiencyInfo {...EFFICIENCY_MOCK} />);

    expect(container).toMatchSnapshot();
  });
});
