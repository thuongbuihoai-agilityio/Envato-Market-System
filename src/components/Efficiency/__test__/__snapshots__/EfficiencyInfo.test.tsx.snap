// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EfficiencyInfo component should render correctly with light mode 1`] = `
<div>
  <div
    class="css-znl6z1"
  >
    <div
      class="css-wax5jp"
    >
      <div>
        EfficiencyInfo component
      </div>
      <div
        class="css-0"
      >
        <div
          class="css-jmbq56"
        >
          <div
            class="css-1qofolz"
          >
            <p
              class="chakra-text css-ss2gmn"
            >
              $
              5
            </p>
            <svg
              fill="none"
              height="12"
              viewBox="0 0 14 12"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                fill="#22C55E"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                fill="#22C55E"
                fill-rule="evenodd"
                opacity="0.4"
              />
            </svg>
          </div>
          <p
            class="chakra-text css-jz5bk3"
          >
            Arrival
          </p>
        </div>
        <div
          class="css-0"
        >
          <div
            class="css-1qofolz"
          >
            <p
              class="chakra-text css-0"
            >
              $
              6
            </p>
            <svg
              fill="none"
              height="12"
              viewBox="0 0 14 12"
              width="14"
            >
              <path
                clip-rule="evenodd"
                d="M10.7749 0.558058C10.5309 0.313981 10.1351 0.313981 9.89107 0.558058L7.39107 3.05806C7.14699 3.30214 7.14699 3.69786 7.39107 3.94194C7.63514 4.18602 8.03087 4.18602 8.27495 3.94194L9.70801 2.50888V11C9.70801 11.3452 9.98783 11.625 10.333 11.625C10.6782 11.625 10.958 11.3452 10.958 11V2.50888L12.3911 3.94194C12.6351 4.18602 13.0309 4.18602 13.2749 3.94194C13.519 3.69786 13.519 3.30214 13.2749 3.05806L10.7749 0.558058Z"
                fill="#1A202C"
                fill-rule="evenodd"
              />
              <path
                clip-rule="evenodd"
                d="M3.22407 11.4419C3.46815 11.686 3.86388 11.686 4.10796 11.4419L6.60796 8.94194C6.85203 8.69786 6.85203 8.30214 6.60796 8.05806C6.36388 7.81398 5.96815 7.81398 5.72407 8.05806L4.29102 9.49112L4.29101 1C4.29101 0.654823 4.01119 0.375001 3.66602 0.375001C3.32084 0.375001 3.04102 0.654823 3.04102 1L3.04102 9.49112L1.60796 8.05806C1.36388 7.81398 0.968151 7.81398 0.724074 8.05806C0.479996 8.30214 0.479996 8.69786 0.724074 8.94194L3.22407 11.4419Z"
                fill="#1A202C"
                fill-rule="evenodd"
                opacity="0.4"
              />
            </svg>
          </div>
          <p
            class="chakra-text css-jz5bk3"
          >
            Spending
          </p>
        </div>
      </div>
    </div>
    <div
      class="css-r12esa"
    >
      <div
        class="css-n2i4qd"
        dir="row"
      >
        <div
          class="css-1tzz1x1"
        />
        <p
          class="chakra-text css-9n3bbc"
        >
          Goal
        </p>
        <p
          class="chakra-text css-15h4n6s"
        >
          50
          %
        </p>
      </div>
      <div
        class="css-n2i4qd"
        dir="row"
      >
        <div
          class="css-1uk3ab5"
        />
        <p
          class="chakra-text css-9n3bbc"
        >
          Spending
        </p>
        <p
          class="chakra-text css-15h4n6s"
        >
          30
          %
        </p>
      </div>
      <div
        class="css-n2i4qd"
        dir="row"
      >
        <div
          class="css-1crr2gq"
        />
        <p
          class="chakra-text css-9n3bbc"
        >
          Others
        </p>
        <p
          class="chakra-text css-15h4n6s"
        >
          20
          %
        </p>
      </div>
    </div>
  </div>
</div>
`;
